### DISCUSSION TITLE
Edgex-go services are not running in v1.3.1

### DISCUSSION THREAD

**Discussion by HCL-BMC** [edgexfoundry/community#86] [General]
I cloned the code from the below path.
 https://github.com/edgexfoundry/edgex-go

Make build is done and binaries were generated.
Then I ran the below commands in the script to launch.

redis-server &
cd /home/kumar/Kumar/EdgeX/edgex-go/cmd/core-command
export EDGEX_SECURITY_SECRET_STORE=false
./core-command &
cd /home/kumar/Kumar/EdgeX/edgex-go/cmd/core-metadata
export EDGEX_SECURITY_SECRET_STORE=false
./core-metadata &
cd /home/kumar/Kumar/EdgeX/edgex-go/cmd/core-data
export EDGEX_SECURITY_SECRET_STORE=false
./core-data &
cd /home/kumar/Kumar/EdgeX/edgex-go/cmd/support-scheduler
export EDGEX_SECURITY_SECRET_STORE=false
./support-scheduler &
cd /home/kumar/Kumar/EdgeX/edgex-go/cmd/sys-mgmt-agent
export EDGEX_SECURITY_SECRET_STORE=false
./sys-mgmt-agent &
cd /home/kumar/Kumar/EdgeX/edgex-go/cmd/sys-mgmt-executor
export EDGEX_SECURITY_SECRET_STORE=false
./sys-mgmt-executor &
 cd /home/kumar/Kumar/EdgeX/edgex-go/cmd/support-notifications
export EDGEX_SECURITY_SECRET_STORE=false
./support-notifications &

Then I ran core-command binary separately. I got the below error.

level=INFO ts=2023-04-06T13:35:39.68051141Z app=core-command source=secret.go:200 msg="load token from file"
level=WARN ts=2023-04-06T13:35:39.6806438Z app=core-command source=secret.go:124 msg="Retryable failure while creating SecretClient: open /tmp/edgex/secrets/core-command/secrets-token.json: no such file or directory"

Any suggestions to resolve this issue ?

**Comment by bnevis-i** [comment]
You will have better luck running the no-security docker-compose files from https://github.com/edgexfoundry/edgex-compose/

P.S. `export EDGEX_SECURITY_SECRET_STORE=false` multiple times is redundant -- once it is set, it inherits to all subprocesses launch from the same shell.  Also, I couldn't reproduce your error -- when I set the env var, it doesn't try to read secrets-token.json, as designed.

**Reply by HCL-BMC** [reply]
Thanks for your suggestions.

I am not running edgex with docker. Natively running egdex-go in ubuntu 22.04 LTS.

I tried export the variable "export EDGEX_SECURITY_SECRET_STORE=false" one time in the script. 

When I ran core-command binary separately getting this issue and all micro services are not running.
level=WARN ts=2023-04-06T13:35:39.6806438Z app=core-command source=secret.go:124 msg="Retryable failure while creating SecretClient: open /tmp/edgex/secrets/core-command/secrets-token.json: no such file or directory"

**Comment by bnevis-i** [comment]
Can you paste the full log and the configuration/toml/configuration.yaml you are using?  Also, I suggest running either the docker or snap and reverse engineering from there.

**Reply by HCL-BMC** [reply]
You mean configuration.yaml from edgex-go all folders ?  I couldn't find the configuration/toml/configuration.yaml in exdgex-go.

I have attached other configuration.yaml file paths and full logs.
![Screenshot from 2023-04-07 22-18-40](https://user-images.githubusercontent.com/36533514/230646643-a943cfa4-46c8-4811-9240-5f0aaae791fe.png)
[log.txt](https://github.com/edgexfoundry/community/files/11180244/log.txt)

**Reply by cloudxxx8** [reply]
configuration.ymal comes from the `main` branch, which is developing for v3.0.
If you are using v2.1.x, please switch to `v2.1.1` tag for everything.

**Reply by HCL-BMC** [reply]
I switched to Hanoi branch(v1.3.0). Now all the micro services are running now. 
Thanks for your suggestions.

**Comment by lenny-goodell** [comment]
Please upgrade to 2.1 (LTS) or 2.3 (latest release). 1.x releases are no longer supported.

**Reply by HCL-BMC** [reply]
I tried with 2.1(LTS).  But, still edgex micro services are not running for me.
https://github.com/edgexfoundry/edgex-go/tree/v2.1.0
 
make build is done. Build is success and binaries were generated. 
After make build, I started all the service using edged_launch script. I attached the script for reference.

Logs :

level=WARN ts=2023-04-07T15:56:52.960734222Z app=core-data source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"
level=WARN ts=2023-04-07T15:56:53.946605965Z app=core-metadata source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"
level=WARN ts=2023-04-07T15:56:53.948426863Z app=support-scheduler source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"
level=WARN ts=2023-04-07T15:56:53.961538631Z app=core-data source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"
level=WARN ts=2023-04-07T15:56:54.946880039Z app=core-metadata source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"
level=WARN ts=2023-04-07T15:56:54.949447298Z app=support-scheduler source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"
level=WARN ts=2023-04-07T15:56:54.962831631Z app=core-data source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"
level=WARN ts=2023-04-07T15:56:55.947623432Z app=core-metadata source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"
level=WARN ts=2023-04-07T15:56:55.949682451Z app=support-scheduler source=database.go:91 msg="couldn't retrieve database credentials: InsecureSecrets missing from configuration"


netstat logs : 
kumar@kumar-ThinkPad-X1-Yoga-1st:~/Kumar/EdgeX/edgex_2.1.0$ sudo netstat -ntlp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.1:6463          0.0.0.0:*               LISTEN      4505/app.asar --no- 
tcp        0      0 127.0.0.1:40139         0.0.0.0:*               LISTEN      841/containerd      
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      867/sshd: /usr/sbin 
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      829/cupsd           
tcp        0      0 0.0.0.0:6379            0.0.0.0:*               LISTEN      49205/redis-server  
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      653/systemd-resolve 
tcp6       0      0 ::1:631                 :::*                    LISTEN      829/cupsd           
tcp6       0      0 :::22                   :::*                    LISTEN      867/sshd: /usr/sbin 
tcp6       0      0 :::6379                 :::*                    LISTEN      49205/redis-server  

/home/kumar/Kumar/EdgeX/edgex_2.1.0/edgex_launch.sh

**Reply by HCL-BMC** [reply]
file:///home/kumar/Kumar/EdgeX/edgex_2.1.0/edgex_launch.sh

Attached the script for your refference.

**Reply by HCL-BMC** [reply]
[edgex_launch.txt](https://github.com/edgexfoundry/community/files/11180076/edgex_launch.txt)

**Comment by HCL-BMC** [comment]
I switched to Hanoi branch(v1.3.0). Now all the micro services are running now.
Thanks all for your suggestions.
